persistence/jdbc/src/test/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStoreTxFunctionalTest.java

```diff
@@ -0,0 +1,132 @@
+import org.infinispan.transaction.TransactionMode;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+@Test(groups = {"functional", "smoke"}, testName = "persistence.jdbc.stringbased.JdbcStringBasedStoreTxFunctionalTest")
+public class JdbcStringBasedStoreTxFunctionalTest extends JdbcStringBasedStoreFunctionalTest {
+   private boolean transactionalConfig;
+   private boolean sharedConfig;
+
+   JdbcStringBasedStoreTxFunctionalTest transactionalConfig(boolean transactionalConfig) {
+      this.transactionalConfig = transactionalConfig;
+      return this;
+   }
+
+   JdbcStringBasedStoreTxFunctionalTest sharedConfig(boolean sharedConfig) {
+      this.sharedConfig = sharedConfig;
+      return this;
+   }
+
+   @Override
+   protected ConfigurationBuilder getDefaultCacheConfiguration() {
+      ConfigurationBuilder configurationBuilder = super.getDefaultCacheConfiguration();
+      configurationBuilder.transaction().transactionMode(TransactionMode.TRANSACTIONAL);
+      if (sharedConfig) {
+         // Shared requires a clustered config, even though we have a single node
+         configurationBuilder.clustering().cacheMode(CacheMode.REPL_SYNC);
+      }
+      return configurationBuilder;
+   }
+
+   @Override
+   protected void modifyJdbcConfiguration(JdbcStringBasedStoreConfigurationBuilder builder) {
+      builder.transactional(transactionalConfig);
+      builder.shared(sharedConfig);
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager(boolean start, GlobalConfigurationBuilder global,
+         ConfigurationBuilder cb) {
+      // Make sure defaults are transactional as well for created configs
+      if (transactionalConfig) {
+         cb.transaction().transactionMode(TransactionMode.TRANSACTIONAL);
+      }
+      if (sharedConfig) {
+         ConfigurationBuilderHolder holder = new ConfigurationBuilderHolder(Thread.currentThread().getContextClassLoader(), global);
+         if (cb != null) {
+            String defaultName = JdbcStringBasedStoreTxFunctionalTest.class.getName() + "-default";
+            global.defaultCacheName(defaultName);
+            holder.newConfigurationBuilder(defaultName).read(cb.build());
+         }
+         global.transport().defaultTransport();
+         return TestCacheManagerFactory.createClusteredCacheManager(start, holder);
+      } else {
+         return super.createCacheManager(start, global, cb);
+      }
+   }
+
+   @Factory
+   public Object[] factory() {
+      return new Object[]{
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(true).sharedConfig(false),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(false).sharedConfig(false),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(true).sharedConfig(true),
+            new JdbcStringBasedStoreTxFunctionalTest().transactionalConfig(false).sharedConfig(true),
+      };
+   }
```

Comment:
Can we have this at the top of the class so it's not necessary to scroll to see the parameters.

Reply:
Okay.